### PR TITLE
Work in Progress: React Application for Uploading Images and Attaching Metadata

### PR DESCRIPTION
**Pull request by ManvithaPonnapati**

### CODE REVIEW

**Comment by vict0rsch** [comment]
Who's responsible for this?

@DatCorno @Vahe987 @notoraptor ?

**Comment by Vahe987** [comment]
@DatCorno will review and merge

File: reactapp/public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (corneauf) [inline reactapp/public/manifest.json]: Maybe we could change this to look more "official".

File: reactapp/src/components/UploadImages.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+import React, {Component, createRef} from 'react';
+import Dropzone from 'react-dropzone';
+import "../styles/image_uploader.css";
+import {post} from 'axios';
+
+//Setup for dropzone component. createRef is for creating access/reference to the HTML page's DOM
```
Comment (corneauf) [inline reactapp/src/components/UploadImages.js]: I'm not familiar with Javascript and its possible toolchains but, is there a way to create a documentation comment akin to Python's docstrings which we could build and host with Sphinx?

File: reactapp/src/components/UploadImages.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    //Method called upload button is submitted.
+    //e is the input event that is automatically passed for this ui event
+    handleSubmit = (e) => {
+        e.preventDefault() // Stop form submit
+        //Iterate through to submit all files
+        this.props.fileLoader(this.state.files)
+        //upload the first file from the list of smiles(remember state.files is the only you can access the files
+        //from onDrop outside of that method. (example of state functioning like a global variable)
```
Comment (corneauf) [inline reactapp/src/components/UploadImages.js]: Let's remove unused commented code. It's a technical debt and the commit history will store it if we need it.

File: reactapp/src/components/UploadImages.js (hunk 2 above)
Comment (vict0rsch) [reply reactapp/src/components/UploadImages.js]: Maybe look into this https://www.docz.site/

File: reactapp/src/layouts/finalPage.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import {withCookies} from "react-cookie";
+import "../styles/image_uploader.css";
+
+class FinalPage extends React.Component {
+
+    render() {
+
+        return (
+            <div className="upload-container">
+                <h3>Thank you for your contribution</h3>
```
Comment (corneauf) [inline reactapp/src/layouts/finalPage.js]: We should add a TODO for the GDPR compliance.

File: reactapp/src/components/UploadImages.js (hunk 2 above)
Comment (vict0rsch) [reply reactapp/src/components/UploadImages.js]: Or maybe this is more relevant https://milmazz.uno/article/2014/08/27/how-to-document-your-javascript-code/

**Comment by ManvithaPonnapati** [comment]
Great.Let me do another pass with better docs!

File: reactapp/src/components/UploadImages.js (hunk 2 above)
Comment (ManvithaPonnapati) [reply reactapp/src/components/UploadImages.js]: Going with this - https://github.com/styleguidist/react-styleguidist . Seems to be a well supported React Library  which uses react-docgen underneath it

**Comment by notoraptor** [comment]
Hi! Since this code uses Javascript and ReactJS, I will review it in place of @DatCorno !

Hi, @ManvithaPonnapati , I see there are still some TODOs here, and some @DatCorno 's comments are not yet addressed . Do you plan to make new commits about TODOs and/or remaining comments in next days?

Thanks!

**Pull request closed without merging**